__tests__/tests/components/NotificationListItem.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { shallow, render } from 'enzyme';
+import { NotificationListItem } from 'components';
+import { View, TouchableOpacity } from 'react-native';
+
+const defaultProps = {
+  id: 1,
+  notification: {
+    subject: {
+      type: 'Commit',
+    },
+  },
+  iconAction() {},
+  navigationAction() {},
+};
+
+describe('<NotificationListItem />', () => {
+  it("should render a View component if notification type is 'Commit'", () => {
+    const wrapper = shallow(<NotificationListItem {...defaultProps} />);
+
+    const result = wrapper.instance().getComponentType();
+
+    expect(result).toBe(View);
+  });
+
+  it("should render a TouchableOpacity component if notification type is not 'Commit'", () => {
+    const notification = {
+      subject: {
+        type: 'not a commit',
+      },
+    };
+    const wrapper = shallow(
+      <NotificationListItem {...defaultProps} notification={notification} />
+    );
+
+    const result = wrapper.instance().getComponentType();
+
+    expect(result).toBe(TouchableOpacity);
+  });
+
+  it('should call navigation when user press TitleComponent', () => {
```

Comment:
I would say `'should call navigation when TitleComponent is pressed`'. Talking about the user is more appropriate in an E2E test.

Reply:
Actually, I take that back. Talking about the user is fine. If you want to do it that way, just switch "press" to "presses".